### DISCUSSION TITLE
How can I get the `end_byte` of a pattern?  (Opposite of `ts_query_start_byte_for_pattern()`)

### DISCUSSION THREAD

**Discussion by ziontee113** [tree-sitter/tree-sitter#3431] [Q&A] [answered]
Thank you all for the amazing library :heart: 

I wonder of there's a `ts_query_end_byte_for_pattern` out there? Since I can't find it in the API. All I can find is `ts_query_start_byte_for_pattern()`.

Currently in my Zig program, I'd have to:
```zig
    const pattern = zig_syntax_highlighting_patterns;
    const query_pattern_count = query.getPatternCount();
    for (0..query_pattern_count) |i| {
        const start = query.getStartByteForPattern(@as(u32, @intCast(i)));
        const end = if (i < query_pattern_count - 1) query.getStartByteForPattern(@as(u32, @intCast(i + 1))) else pattern.len;
        const current_pattern = pattern[start..end];
        std.debug.print("======================\n{s}\n======================\n", .{current_pattern});
    }
```

to get the full string of the pattern.
This isn't perfect due to the range from one pattern to another might have new lines and comments, I'd have to trim them manually.

Is there a better solution for this? Thank you very much :pray:

**Comment by maxbrunsfeld** [comment, marked as answer]
Right now, we don't record the end bytes of patterns, but I'd be open to adding that.
